src/main/java/teammates/ui/servlets/HibernateSessionRequestFilter.java

```diff
@@ -0,0 +1,39 @@
+package teammates.ui.servlets;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import teammates.common.util.HibernateUtil;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+
+public class HibernateSessionRequestFilter implements Filter {
+
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        sessionFactory = HibernateUtil.getSessionFactory();
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws ServletException, IOException {
```

Comment:
It would be good for this method to handle rollbacks on error so that we don't have to add rollback code everywhere else. We should also add the log messages after the transaction is committed, though I'm not sure if we are able to customise log messages this way.

Reply:
> It would be good for this method to handle rollbacks on error so that we don't have to add rollback code everywhere else. 

Ideally yes. I did this originally, only to realise in the `CreateCourseAction` class `InvalidOperationException` and `InvalidHttpRequestBodyException` thrown (e.g. due to duplicate record found) does not propagate up to the filter, which means if we try to catch it in the filter it will never get caught. However, I do think it is still worth looking to see if there's a way to handle Hibernate errors and perform rollbacks at one place if possible so that we don't have to add rollback code everywhere. I'll leave it as an open problem in issue #20.